nx/lib/nx/lin_alg.ex

```diff
@@ -1199,12 +1199,18 @@ defmodule Nx.LinAlg do
     {p, l, u} = Nx.LinAlg.lu(t)
 
     diag = Nx.take_diagonal(l) * Nx.take_diagonal(u)
-
     is_zero = Nx.any(diag == 0)
+    transitions = Nx.dot(p, Nx.iota({n}))
 
-    iota = Nx.iota({n})
+    parity =
+      transform(transitions, fn transitions ->
+        {m} = Nx.shape(transitions)
 
-    parity = Nx.sum(Nx.dot(p, iota) != iota)
+        for i <- 0..(m - 1), reduce: 0 do
+          parity ->
+            Nx.add(parity, Nx.sum(Nx.greater(transitions[i], transitions[i..-1//1])))
```

Comment:
Looks good! I'll try to find a non-transform implementation on this

Reply:
> I had opened an issue on this earlier this week :)
I've started it on your report. Couldn't figure out how to handle it without transform, unfortunately.

Reply:
I can't properly make a code suggestion here, but this is what I got:
```elixir
   transitions = Nx.dot(p, Nx.iota({n}))

    lower_tri_mask = {n, n} |> Nx.iota(axis: 0) |> Nx.greater(Nx.iota({n, n}, axis: 1))

    tiled_transitions = Nx.tile(transitions, [n, 1])

    tiled_transitions_with_mask = Nx.select(lower_tri_mask, n, tiled_transitions)

    parity =
      tiled_transitions
      |> Nx.transpose()
      |> Nx.greater(tiled_transitions_with_mask)
      |> Nx.sum()
```

Reply:
This will use a bit more memory, but should be faster

Reply:
This is the overall idea:

```elixir
iex(29)> Nx.tile(Nx.reshape(transitions, {4, 1}), [1, 4]) |> Nx.greater(tiled)
#Nx.Tensor<
  u8[4][4]
  [
    [0, 0, 0, 0],
    [0, 0, 1, 1],
    [0, 0, 0, 1],
    [0, 0, 0, 0]
  ]
>
iex(30)> for i<-0..3, do: Nx.greater(transitions[i], transitions[i..-1//1])                                                   [              
  #Nx.Tensor<
    u8[4]
    [0, 0, 0, 0]
  >,
  #Nx.Tensor<
    u8[3]
    [0, 1, 1]
  >,
  #Nx.Tensor<
    u8[2]
    [0, 1]
  >,
  #Nx.Tensor<
    u8[1]
    [0]
  >
]
```

Second result is what your code generates, first result is the last step before Nx.sum() in my snippet above

Reply:
Multiply and select should be roughly the same speed. I had chosen select to ensure greater would fail by inserting a value greater than all others. If this works, it works!